cpp/ql/src/semmle/code/cpp/models/implementations/Allocation.qll

```diff
@@ -255,6 +255,24 @@ class OperatorNewAllocationFunction extends AllocationFunction {
   }
 }
 
+pragma[inline]
+private predicate deconstructSizeExpr(Expr sizeExpr, Expr lengthExpr, int sizeof) {
```

Comment:
Why `pragma[inline]`?

Reply:
The idea is to not compute the predicate on all Exprs (the second case already includes almost all expressions), but only for those expressions that are the size argument to an AllocationExpr.
If you don't like this, I can remove it.

Reply:
We generally prefer to let the optimizer make decisions like this unless the result is demonstrably performing worse than it should be.  This keeps the QL clean and easy to maintain.

That said, if removing `pragma[inline]` drastically increases the time spend computing this (or a related) predicate, then lets keep it.

Reply:
:+1: